src/main/java/kafka/vertx/demo/Main.java

```diff
@@ -35,8 +48,56 @@ public static void main(String[] args) {
     Future<String> periodicProducerDeployment = vertx.deployVerticle(new PeriodicProducer());
     Future<String> webSocketServerDeployment = vertx.deployVerticle(new WebSocketServer());
 
-    CompositeFuture.all(periodicProducerDeployment, webSocketServerDeployment)
+    CompositeFuture.join(periodicProducerDeployment, webSocketServerDeployment)
       .onSuccess(ok -> logger.info("✅ Application started in {}ms", currentTimeMillis() - startTime))
       .onFailure(err -> logger.error("❌ Application failed to start", err));
   }
+
+  public static Future<JsonObject> loadKafkaConfig(Vertx vertx, String propertiesPath) {
+    File propertiesPathFile = new File(propertiesPath);
+    String properties = propertiesPathFile.isDirectory() ? new File(propertiesPathFile, DEFAULT_PROPERTIES_PATH).toString() : propertiesPath;
+    ConfigRetriever configRetriever =  ConfigRetriever.create(vertx,
+      new ConfigRetrieverOptions().addStore(
+        new ConfigStoreOptions()
+          .setType("file")
+          .setFormat("properties")
+          .setConfig(new JsonObject().put("path", properties).put("raw-data", true)))).setConfigurationProcessor(configurationProcessor(properties));
+    FileSystem fileSystem = vertx.fileSystem();
+    return fileSystem.exists(properties)
+      .compose(exists -> {
+        if (exists) {
+          return configRetriever.getConfig();
+        } else {
+          return Future.failedFuture(String.format("Kafka properties file is missing. Either specify using -D%s=<path> or use the default path of %s.", PROPERTIES_PATH_ENV_NAME, DEFAULT_PROPERTIES_PATH));
+        }
+      });
+  }
+
+  public static Function<JsonObject, JsonObject> configurationProcessor(String propertiesPath) {
+    Path propertiesAbsolutePath = new File(propertiesPath).toPath().toAbsolutePath();
```

Comment:
More sensible to pass the file in to avoid having to create a new file? Unless File is such a small class footprint it's not a problem creating many?